data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ConditionalExpressionEvaluatorIT.java

```diff
@@ -0,0 +1,113 @@
+        final ConditionalExpressionEvaluator evaluator = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+
+        final Boolean actual = evaluator.evaluate(expression, event);
+
+        assertThat(actual, is(expected));
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidExpressionArguments")
+    void testConditionalExpressionEvaluatorThrows(final String expression, final Event event, final Class<? extends Throwable> throwable) {
+        final ConditionalExpressionEvaluator evaluator = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+
+        assertThrows(throwable, () -> evaluator.evaluate(expression, event));
+    }
+
+    private static Stream<Arguments> validExpressionArguments() {
+        return Stream.of(
+                Arguments.of("true", event("{}"), true),
+                Arguments.of("/status_code == 200", event("{\"status_code\": 200}"), true),
+                Arguments.of("/status_code != 300", event("{\"status_code\": 200}"), true),
+                Arguments.of("/success == /status_code", event("{\"success\": true, \"status_code\": 200}"), false),
+                Arguments.of("/success != /status_code", event("{\"success\": true, \"status_code\": 200}"), true),
+                Arguments.of("/pi == 3.14159", event("{\"pi\": 3.14159}"), true),
+                Arguments.of("true == (/is_cool == true)", event("{\"is_cool\": true}"), true),
+                Arguments.of("not /is_cool", event("{\"is_cool\": true}"), false),
+                Arguments.of("/status_code < 300", event("{\"status_code\": 200}"), true),
+                Arguments.of("/status_code <= 0", event("{\"status_code\": 200}"), false),
+                Arguments.of("/status_code > 0", event("{\"status_code\": 200}"), true),
+                Arguments.of("/status_code >= 300", event("{\"status_code\": 200}"), false),
+                Arguments.of("-/status_code == -200", event("{\"status_code\": 200}"), true),
+                Arguments.of("/success and /status_code == 200", event("{\"success\": true, \"status_code\": 200}"), true),
+                Arguments.of("/success or /status_code == 200", event("{\"success\": false, \"status_code\": 200}"), true),
+                Arguments.of("(/success == true) or (/status_code == 200)", event("{\"success\": false, \"status_code\": 200}"), true),
+                Arguments.of("/should_drop", event("{\"should_drop\": true}"), true),
+                Arguments.of("/should_drop", event("{\"should_drop\": false}"), false),
+                Arguments.of("/logs/2/should_drop", event("{\"logs\": [{}, {}, {\"should_drop\": true}]}"), true),
+                Arguments.of("\"/\"complex\" ~1~0json~1 'key'\" == true", event("{\"\\\"complex\\\"  /~json/ 'key'\": true}"), true)
+        );
+    }
+
+    private static Stream<Arguments> invalidExpressionArguments() {
+        return Stream.of(
+                Arguments.of("/missing", event("{}"), RuntimeException.class),
```

Comment:
It seems the exception is always `RuntimeException`. Maybe we don't need this as a parameter into the tests.